api/src/main/java/net/md_5/bungee/api/event/PostLoginEvent.java

```diff
@@ -20,4 +21,11 @@
      * The player involved with this event.
      */
     private final ProxiedPlayer player;
+    private ServerInfo joinTarget = null;
+
+    public void setTarget(ServerInfo serverInfo) {
+        joinTarget = serverInfo;
+    }
+
+    public ServerInfo getTarget() { return joinTarget; }
```

Comment:
If you simply name the variable ServerInfo ```target``` you can omit the getter and the setter due to lombok.

Currently lombok generated ```getJoinTarget()``` and ```setJoinTarget()``` methods.